utils/src/main/java/org/corfudb/utils/lock/LockClient.java

```diff
@@ -154,6 +183,30 @@ private void monitorLocks() {
         ));
     }
 
+
+    /**
+     * Stop monitoring the locks.
+     */
+    private void stopMonitorLocks() {
+        synchronized (lockMonitorFuture) {
+            if (lockMonitorFuture.isPresent() && !lockMonitorFuture.get().isDone()) {
+                boolean cancelled = lockMonitorFuture.get().cancel(false);
+                if (!cancelled) {
+                    log.error("Lock: Could not cancel the future for lease renewal!!!", lock.getLockId());
+                }
+            }
+            log.warn("Lock: canceled the future for monitor locks!!!", lock.getLockId());
+        }
+    }
+
+
+    public void stop() {
+        lock.input(LockEvent.UNRECOVERABLE_ERROR);
+        taskScheduler.shutdownNow();
```

Comment:
why do we call shutdown()?  a clean way to shutdown the executorservice is to call shutdown() and then shutdownnow() - https://www.baeldung.com/java-executor-service-tutorial

Reply:
I did shutdown() first, it generated a lot of exceptions. For this case, shutdownnow() seems work better.